arch/arm/src/stm32/stm32_eth.c

```diff
@@ -3430,11 +3430,14 @@ static inline void stm32_ethgpioconfig(FAR struct stm32_ethmac_s *priv)
 #if defined(CONFIG_STM32_MII) || defined(CONFIG_STM32_RMII)
 
   /* MDC and MDIO are common to both modes */
```

Comment:
@garfieldG - Since you are closer to the problem, I need your help to understand this change.

What was the problem you are trying to solve? What it different in your HW and why are this changes needed?

Reply:
@garfieldG - I am starting. to get the picture. The custom board you have has a PHY that is supported in nuttx but not wired the same. 

Does it uses RMMI?  Where is the SPI code?

Reply:
Yes, Ethernet PHY (usually) has 2 communication links to the MAC (the CPU hardware). Usually the 100Mbit/sec data is over MII protocol (either RMII, MII or GMII). but the control (on/off, 100mbit/10mbit, HD/FD and so on) is done by other lines, the default is MDIO standard. It seems like Nuttx handles this standard (you can see it in stm32_ethernet code). 
When the PHY is a switch MDIO is less relevant, so you need other communication. In our case its a complicated protocol over SPI lines.
The SPI code is not relevant, because the two protocols are independent. We actually don't do much over the SPI, just monitoring the link (up/down and so on)

Reply:
@davids5

Reply:
@garfieldG - I am not familiar enough with what you are trying to accomplish.   What I was trying and failing at understanding so far, is: Is the change you need to make to a driver in NuttX' stmf7 code base a change specific to just your HW are is it something that would benefit upstream Nuttx? For example, a bug fix or a set of CONFIG_xxx that are wrong in upstream. 

We can have a call to discuss this in real time if need be. Skype:david_s5